### PR TITLE
macros: add register vec macros

### LINKED ISSUE #35: macro for registering vec metric

### PR DESCRIPTION
**Pull request by overvenus**
Resolved #35 

@siddontang @disksing

### CODE REVIEW

File: src/macros.rs [hunk 1]
```diff
@@ -105,6 +105,23 @@ macro_rules! register_counter {
 }
 
 #[macro_export]
+macro_rules! register_counter_vec {
+    ( $ OPTS : expr , $ LABELS_NAMES : expr ) => {
```
Comment (siddontang) [inline src/macros.rs]: should support simple `(name, help, label_names)`

File: src/macros.rs [hunk 2]
```diff
@@ -105,6 +105,22 @@ macro_rules! register_counter {
...
+        {
+            let counter_vec = $crate::CounterVec::new($OPTS, $LABELS_NAMES).unwrap();
+            $crate::register(Box::new(counter_vec.clone())).map(|_| counter_vec)
+        }
+    };
+
+    ( $ NAME : expr , $ HELP : expr , $ LABELS : expr , $ LABELS_NAMES : expr ) => {
```
Comment (siddontang) [inline src/macros.rs]: do we must pass const labels ? I think no need.

Comment (overvenus) [inline src/macros.rs]: Just pass `labels!{}` for omitting const_labels.

I think it is pretty confusing if we support 
`( $ NAME : expr , $ HELP : expr , $ LABELS : expr , $ LABELS_NAMES : expr )` and
`( $ NAME : expr , $ HELP : expr , $ LABELS_NAMES : expr )` as the same time.


It is too easy for someone to write something like `register_counter_vec!("name", "help", labels!{"a" => "b",})`

Comment (siddontang) [inline src/macros.rs]: why confusing? you can remove labels here, or rename LABELS to CONST_LABELS and LABELS_NAMES to VARIABLE_NAMES

**Comment by siddontang** [comment]
Think we can merge #38 first and then re-update the comment for the new macros.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**